operations_on_datastructures/inorder_successor_of_bst.cpp

```diff
@@ -0,0 +1,407 @@
+/**
+ * @file
+ * @brief An implementation for finding the [Inorder successor of a binary
+ * search tree](https://www.youtube.com/watch?v=5cPbNCrdotA&t=904s) Inorder
+ * successor of a node is the next node in Inorder traversal of the Binary Tree.
+ * Inorder Successor is NULL for the last node in Inorder traversal.
+ *
+ * ### Case 1: The given node has the right node/subtree
+ *
+ *      In this case the left most deepest node in the right subtree will come
+ * just after the given node as we go to left deep in inorder.
+ *      - Go deep to left most node in right subtree.
+ *        OR, we can also say in case if BST, find the minimum of the subtree
+ * for a given node.
+ *
+ * ### Case 2: The given node does not have a right node/subtree
+ *
+ * #### Method 1: Use parent pointer (store the address of parent nodes)
+ *      If a node does not have right subtree, and we already visited the node
+ * itself, then the next node will be its parent node according to inorder
+ * traversal, and if we are going to parent from left, then the parent would be
+ * unvisited. In other words, go to the nearest ancestor for which given node
+ * would be in left subtree.
+ *
+ * #### Method 2: Search from the root node
+ *      In case if there is no link to the parent, we need to walk the tree
```

Comment:
```suggestion
 *      * In case if there is not linked to the parent, we need to walk to the tree
```

Reply:
I guess this one is fine 🤔, apart from the bullet point part. That's what I wanted to convey but I'll rephrase it as, 

```in case if there is no link from a child node to the parent node, we need to walk down the tree...```